src/redis_bitmap.cc

```diff
@@ -313,6 +315,191 @@ rocksdb::Status Bitmap::BitPos(const Slice &user_key, bool bit, int64_t start,
   return rocksdb::Status::OK();
 }
 
+rocksdb::Status Bitmap::BitOp(BitOpFlags op_flag, const std::string &op_name,
+                              const Slice &user_key, const std::vector<Slice> &op_keys, int64_t *len) {
+  std::string ns_key, raw_value, ns_op_key;
+  AppendNamespacePrefix(user_key, &ns_key);
+  LockGuard guard(storage_->GetLockManager(), ns_key);
+
+  rocksdb::Status s;
+  std::vector<std::pair<std::string, BitmapMetadata>> meta_pairs;
+  uint64_t max_size = 0, num_keys = op_keys.size();
+
+  for (const auto &op_key : op_keys) {
+    BitmapMetadata metadata(false);
+    AppendNamespacePrefix(op_key, &ns_op_key);
+    s = GetMetadata(ns_op_key, &metadata, &raw_value);
+    if (!s.ok()) {
+      if (s.IsNotFound()) {
+        num_keys--;
+        continue;
+      }
+      return s;
+    }
+    if (metadata.Type() == kRedisString) {
```

Comment:
Why not consider the type `BitmapString`?

Reply:
When using setbit on the string type, master-slave replication will have a tremendous amount of write amplification because Kvrocks replication does not use command propagation like Redis. In addition, using type bitmap conserves storage space. So we recommend using type bitmap.

Reply:
Maybe in the future we can disable `BitmapString` by default in kvrocks unless the user modifies the kvrocks.conf. HDYT @ChrisZMF  @git-hulk

Reply:
Maybe in the future, operations between `BitmapString` can be supported separately, which is not difficult. If you want to support the operation between `bitmap` and `string`, the logic will become very complicated. It seems unnecessary.

Reply:
Got it, thx. It can be considered in the future according to user needs.